course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,164 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){}
+
+MyString::MyString(const char* argString)
+{
+        buffer = std::unique_ptr<char[]>(new char[ std::strlen(argString) + 1]);
+        std::strcpy(buffer.get(), argString);;
```

Comment:
Do you have compilation warnings enabled? Compiler should warn about the two semicolons...